src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2346,6 +2348,42 @@ private function registerHttpClientConfiguration(array $config, ContainerBuilder
             $this->registerRetryableHttpClient($retryOptions, 'http_client', $container);
         }
 
+        if ($hasUriTemplate = $config['uri_template']['enabled']) {
+            if (!class_exists(UriTemplateHttpClient::class)) {
+                throw new LogicException('Support for URI template requires symfony/http-client 6.3 or higher, try upgrading.');
+            }
+
+            if ($expanderId = $config['uri_template']['expander'] ?? null) {
+                $uriTemplateExpander = new Reference($expanderId);
+            } elseif (class_exists(\Rize\UriTemplate::class)) {
```

Comment:
Checking it here avoids runtime check by directly injecting a `\Closure` when possible :sweat_smile: